meson: HAVE_ENVIRON_DECL is being defined despite not being available on macOS

Despite Meson properly detecting that `environ` is not defined on macOS, it still defines the macro `HAVE_ENVIRON_DECL` to `0` due to the use of [cfg_data.set10()](https://mesonbuild.com/Reference-manual_returned_cfg_data.html#cfg_dataset10). It should use `cfg_data.set()` instead to avoid defining it. The value is unimportant, only whether it is defined, as that is how the macro is used.

```
...
Header "unistd.h" has symbol "environ" : NO 
...
FAILED: lib/libcommon.a.p/env.c.o 
clang -Ilib/libcommon.a.p -Ilib -I../src/lib -Iinclude -I../src/include -fdiagnostics-color=always -DNDEBUG -Wall -Winvalid-pch -std=c99 -O3 -D_GNU_SOURCE -include /Users/jenkins/workspace/prod-v1/bsr/11752/ecfca/.conan/data/util-linux-libuuid/2.40/_/_/build/6280371def85df4402bdeba403e98e478874576a/build-release/config.h -fno-common -Waddress-of-packed-member -Wembedded-directive -Wextra-semi -Wformat-security -Wimplicit-function-declaration -Wmissing-declarations -Wmissing-prototypes -Wnested-externs -Wno-missing-field-initializers -Wold-style-definition -Wpointer-arith -Wredundant-decls -Wsign-compare -Wstrict-prototypes -Wtype-limits -Wuninitialized -Wunused-parameter -Wunused-result -Wunused-variable -Wvla -Walloca -isysroot /Applications/conan/xcode/13.0/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk -arch arm64 -MD -MQ lib/libcommon.a.p/env.c.o -MF lib/libcommon.a.p/env.c.o.d -o lib/libcommon.a.p/env.c.o -c ../src/lib/env.c
In file included from ../src/lib/env.c:22:
../src/include/all-io.h:117:40: warning: unused parameter 'out' [-Wunused-parameter]
static inline ssize_t sendfile_all(int out, int in, off_t *off, size_t count)
                                       ^
../src/include/all-io.h:117:49: warning: unused parameter 'in' [-Wunused-parameter]
static inline ssize_t sendfile_all(int out, int in, off_t *off, size_t count)
                                                ^
../src/include/all-io.h:117:60: warning: unused parameter 'off' [-Wunused-parameter]
static inline ssize_t sendfile_all(int out, int in, off_t *off, size_t count)
                                                           ^
../src/include/all-io.h:117:72: warning: unused parameter 'count' [-Wunused-parameter]
static inline ssize_t sendfile_all(int out, int in, off_t *off, size_t count)
                                                                       ^
../src/lib/env.c:149:23: error: use of undeclared identifier 'environ'
        char **envp = environ;
                      ^
4 warnings and 1 error generated.
...
```

--- a/meson.build
+++ b/meson.build
@@ -456,7 +456,7 @@ have_cpu_set_t = cc.sizeof('cpu_set_t', prefix : '#define _GNU_SOURCE\n#include 
 conf.set('HAVE_CPU_SET_T', have_cpu_set_t ? 1 : false)
 
 have = cc.has_header_symbol('unistd.h', 'environ', args : '-D_GNU_SOURCE')
-conf.set10('HAVE_ENVIRON_DECL', have)
+conf.set('HAVE_ENVIRON_DECL', have ? 1 : false)
 
 have = cc.has_header_symbol('signal.h', 'sighandler_t', args : '-D_GNU_SOURCE')
 conf.set('HAVE_SIGHANDLER_T', have ? 1 : false)

